src/util/macros.rs

```diff
@@ -143,3 +143,14 @@ macro_rules! thd_name {
         }).unwrap_or_else(|| $name.to_owned())
     });
 }
+
+/// Simulating go's defer.
+///
+/// Please note that, different from go, this defer is bound to scope.
+#[macro_export]
+macro_rules! defer {
+    ($t:expr) => (
+        let ctx = $crate::util::DeferContext::new(|| $t);
```

Comment:
`Drop` will be invoked immediately then. But it's expected to be invoked when scope is about to being destroyed.